autoload/neomake.vim

```diff
@@ -260,9 +271,10 @@ function! s:AddExprCallback(maker) abort
 
         " On the first valid error identified by a maker,
         " clear the existing signs in the buffer
-        call s:CleanSignsAndErrors()
+        call s:CleanSignsAndErrors(entry.bufnr)
```

Comment:
I am not sure yet about how `s:last_placed_signs` works. At first glance my answer is yes, it can't (I have to test it to be really sure), but that would really be a shame, since a useful use case is to run a linter with `:Neomake` on every write while a long build with `:Neomake!` is done in the background.

I do not see the difference between `:Neomake`  and `:Neomake!` when it comes to interfering though (what if I run concurrently `:Neomake ghcmod` and `:Neomake hlint`?)

Since there is a guarantee that there will be only one job by maker, we could track signs by maker instead, and remove only the signs corresponding to a given maker.